File: algorithms/Yandex/algorithms_training_4/warm_up/C++/E.cpp
#include <iostream>

using std::cin, std::cout;


void middleLev(const short* arr, int SIZE) {
    unsigned int* lev_arr {new unsigned int[SIZE]};
    unsigned int right_s = 0, left_s = 0;
    for (int i = 0; i != SIZE; ++i) right_s += arr[i];
    lev_arr[0] = right_s - arr[0] * SIZE;
    right_s -= arr[0];
    cout << lev_arr[0];
    for (int i = 1; i != SIZE; ++i) {
        left_s += i * (arr[i] - arr[i - 1]);
        lev_arr[i] = left_s + right_s - arr[i] * (SIZE - i);
        right_s -= arr[i];
        cout << ' ' << lev_arr[i];
    }
    delete[] lev_arr;
}


int inputE() {
    int SIZE;
    cin >> SIZE;
    short* arr {new short[SIZE]};
    for (int i = 0; i != SIZE; ++i) cin >> arr[i];
    middleLev(arr, SIZE);
    delete[] arr;
}

File: algorithms/Yandex/algorithms_training_4/warm_up/Python/E.py
def middleLev(arr: list) -> list:
    lev_arr = [0] * len(arr)
    left_s, right_s = 0, sum(arr)
    lev_arr[0] = right_s - arr[0] * (len(arr))
    right_s -= arr[0]
    for i in range(1, len(arr)):
        left_s += i * (arr[i] - arr[i - 1])  # все предыдущие элементы меньше по модулю
        lev_arr[i] = left_s + right_s - arr[i] * (len(arr) - i)  # вычитаем из всех, включая текущий
        right_s -= arr[i]
    return lev_arr

input()
print(*middleLev(list(map(int, input().split()))))
